src/tile.cpp

```diff
@@ -464,6 +470,26 @@ void Tile::onRemoveTileItem(const SpectatorHashSet& spectators, const std::vecto
 	for (Creature* spectator : spectators) {
 		spectator->onRemoveTileItem(this, cylinderMapPos, iType, item);
 	}
+
+	if (!hasFlag(TILESTATE_PROTECTIONZONE)) {
+		auto it = getItemList();
+		if (it->size() == 0) { // it->empty() does not work for some strange reason here
```

Comment:
Hey this has to be debugged. Deep inside stdlib `empty()` is written as `return size() == 0` so they must be the same.

Reply:
"Can't select the function, due to non necessary rights" not sure why this is throwing as we use the same method for Creature* vectors aswell and it works there.

Reply:
You need to forward the function you want to use here: https://github.com/otland/forgottenserver/blob/master/src/tile.h#L81 because we're using private inheritence